curvefs/src/client/fuse_volume_client.cpp

```diff
@@ -142,6 +142,22 @@ void FuseVolumeClient::FuseOpDestroy(void *userdata) {
     return;
 }
 
+CURVEFS_ERROR FuseVolumeClient::FuseOpOpen(fuse_req_t req, fuse_ino_t ino,
```

Comment:
Does `FuseS3Client::FuseOpOpen` and `FuseVolumeClient::FuseOpen` are only different in truncate ?
If so, you can leave base class's `FuseOpOpen`, add a private virtual member function, e.g., `Truncate`, and call this function if open flags with O_TRUNC.
For the `Trunate` function, the derived class must override it, and do it's own work.

Reply:
> Does `FuseS3Client::FuseOpOpen` and `FuseVolumeClient::FuseOpen` are only different in truncate ?
> If so, you can leave base class's `FuseOpOpen`, add a private virtual member function, e.g., `Truncate`, and call this function if open flags with O_TRUNC.
> For the `Trunate` function, the derived class must override it, and do it's own work.

fixed